### DISCUSSION TITLE
Multiple domain names with single ports vs multiple ports with single domain name?

### DISCUSSION THREAD

**Discussion by jon-can** [gravitational/teleport#5542] [Q&A] [answered]
Can Teleport be configured such that it can use multiple domain names with single ports vs the current design that uses multiple ports with a single domain name? Example: instead of test.example.com:3026 and test.example.com:3080, could we do proxykube.test.example.com and proxyweb.test.example.com? I understand something will have to do the 443 -> right port translation for the pods, more curious if clients would be able to handle this modification in the interaction.

**Comment by klizhentas** [comment, marked as answer]
It is possible to split web and ssh ports like that, but not with kubernetes at the moment. 

You need to set `public_addr`, `ssh_public_addr` and `tunnel_public_addr` to different domain names and `tsh` will use those domains when logging in. We'd need to expand this to kubernetes and app services as well. Can you submit a feature request so we can track this?

**Reply by jon-can** [reply]
Yes sir will do, thank you.

**Reply by awly** [reply]
Note: you can use different domain names for web and kube endpoints.
You just can't reuse the same port (yet).

**Reply by klizhentas** [reply]
@awly This is great news, I tried to find the `kube_public_addr` setting in config file code and did not, can you point to the doc/place where is it located?

**Reply by awly** [reply]
We (prematurely) removed it from the docs in 5.0, but you can still use the old `kubernetes.public_addr` field in `proxy_service`:
```yaml
proxy_service:
  kubernetes:
    enabled: true
    listen_addr: "0.0.0.0:3026"
    public_addr: "proxykube.test.example.com:3026"
```
https://github.com/gravitational/teleport/blob/3458976501d182c8ddd0fb0dc6084b01461b0cee/lib/config/fileconf.go#L1047-L1048
